x/blob/client/cli/payforblob.go

```diff
@@ -63,11 +67,22 @@ func CmdPayForBlob() *cobra.Command {
 	}
 
 	flags.AddTxFlagsToCmd(cmd)
+	cmd.PersistentFlags().Uint8(FlagNamespaceVersion, 0, "Specify the namespace version")
 	cmd.PersistentFlags().Uint8(FlagShareVersion, 0, "Specify the share version")
-
 	return cmd
 }
 
+func getNamespace(namespaceID []byte, namespaceVersion uint8) (appns.Namespace, error) {
+	switch namespaceVersion {
+	case appns.NamespaceVersionZero:
+		nsp := make([]byte, 0, appns.NamespaceSize)
```

Comment:
[nit] 
```suggestion
		nsp := make([]byte, 0, appns.NamespaceIDSize)
```

Reply:
why not namespace size here? aren't we using the full 33?

Reply:
The ID is 32 bytes. The version is 1 byte. The total namespace is 33 bytes. 

Since this variable is just the ID, only 32 bytes are necessary.

Reply:
merged to clear out his PR, but we should still fix this imo if need be,

I'm confused a bit

so we should not be appending the version at all here since its added when we call Bytes()? therefore we're adding it twice?